src/Paket/Program.fs

```diff
@@ -741,6 +743,22 @@ let why (results: ParseResults<WhyArgs>) =
 
     Why.ohWhy(packageName, directDeps, lockFile, groupName, results.Parser.PrintUsage(), options)
 
+let restriction (results: ParseResults<RestrictionArgs>) =
+    let restrictionRaw = results.GetResult <@ RestrictionArgs.Restriction @>
+    let restriction, parseProblems = Requirements.parseRestrictions restrictionRaw
+    
+    for problem in parseProblems |> Seq.map (fun x -> x.AsMessage) do
+        Logging.traceWarnfn "Problem: %s" problem
+        
+    printfn "Restriction: %s" restrictionRaw
```

Comment:
please don't use printfn directly. tracefn all the things

Reply:
Ok, but but just to make it clear: This is in fact program output no logging